src/main/scala/rules/Evaluator.scala

```diff
@@ -292,7 +292,7 @@ object evaluator extends EvaluationRules {
       case ast.Let(x, e0, e1) =>
         eval(s, e0, pve, v)((s1, t0, v1) => {
           val t = v1.decider.appliedFresh("letvar", v1.symbolConverter.toSort(x.typ), s1.relevantQuantifiedVariables)
-          v1.decider.assume(t === t0)
+          v1.decider.assumeDefinition(t === t0)
           val newFuncRec = s1.functionRecorder.recordFreshSnapshot(t.applicable.asInstanceOf[Function])
```

Comment:
Once this PR was accepted, it would be worthwhile to work out if the newly-recorded definitions partly supersedes what the function recorder bookkeeps. If so, the latter could probably be simplified.

Reply:
Yes, makes sense. I'll put that on the list. Thanks!